website/generate-jsdoc.js

```diff
@@ -26,33 +26,49 @@ function generateMarkdown(inputFile) {
     .join("\n");
   fs.writeFileSync(
     path.join(outputDir, `${className}.md`),
-    prettier.format(markdownOutput, { parser: "markdown" })
+    prettier.format(markdownOutput, { parser: "mdx" })
   );
 }
 
 function documentItem(data) {
-  return `## \`${data.id}\`
+  return `${documentHeader(data)}
 
 ${data.examples ? documentExamples(data) : ""}
 
 ${data.description || ""} 
 
+${data.deprecated ? documentDeprecated(data) : ""}
+
 ${data.params ? documentParams(data) : ""}
 
 ${data.returns ? documentReturn(data) : ""}
 `;
 }
 
+function documentHeader(data) {
+  if (data.deprecated)
+    return `## ~~\`${data.id}\`~~`
+  return `## \`${data.id}\``
+}
+
 function documentExamples(data) {
   return `\`\`\`
 ${data.examples}
 \`\`\`
 `;
 }
 
+function documentDeprecated(data) {
+  return `<div style={{ padding: '10px', backgroundColor: '#ff4c59' }}>
```

Comment:
how is this affected by the dark/light modes ( e.g will the dark mode cause text here to appear white)? It looks like this background flags a small warning on wcag color difference with black text and fails AA criteria

Reply:
Updated it, I think it should be good for AA standards now

- Dark: https://webaim.org/resources/contrastchecker/?fcolor=000000&bcolor=E71829

- Light: https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=E71829